gk/fib.c

```diff
@@ -93,12 +157,14 @@ neigh_get_ether_cache_locked(struct neighbor_hash_table *neigh,
 		return eth_cache;
 	}
 
-	ret = rte_hash_add_key(neigh->hash_table, &addr->ip);
+	eth_cache = get_new_ether_cache_locked(neigh, addr, iface, lcore_id);
+	if (eth_cache == NULL)
+		return NULL;
+
+	ret = rte_hash_add_key_data(neigh->hash_table, &addr->ip, eth_cache);
 	if (ret >= 0) {
```

Comment:
The source code actually looks to me like `rte_hash_add_key_data()` returns 0 when the entry is successfully added:

http://dpdk.org/browse/dpdk/tree/lib/librte_hash/rte_cuckoo_hash.c#n702

That's the current version of the code, but the DPDK version we're using does the same thing. So I think the documentation is right, and this test should check `if (ret == 0)`.

I don't think this change affects the code that's now in `neigh_del_ether_cache()`, which is where the original discussion about `rte_hash_add_key_data()` came up.